python/test/segyio_c.py

```diff
@@ -443,30 +443,69 @@ def mkempty():
 
 @tmpfiles(testdata / 'trace-header-extensions.sgy')
 def test_read_and_write_traceheader(tmpdir):
-    f = get_instance_segyfile(tmpdir, 'trace-header-extensions.sgy')
+    field = 1
+    start, stop, step = 0, 2, 1
+    indices = numpy.asarray([0, 1], dtype=numpy.intc)
+    attrs = numpy.empty(len(indices), dtype=numpy.intc)
+    trace_index = 0
 
     def mkempty():
         return bytearray(_segyio.thsize())
 
-    field = 1
-    # note: for the moment all headers are processed using default standard
-    # traceheader mapping, which is incorrect. Test will change in the future
-    # when get-set field functions are updated.
+    def verify_traceheader(f, traceheader_index, type, expected):
+        attrs = numpy.empty(len(indices), dtype=type)
+
+        traceheader = f.getth(trace_index, traceheader_index, mkempty())
+        assert f.getfield(traceheader, traceheader_index, field) == expected[0]
+
+        f.field_forall(attrs, traceheader_index, start, stop, step, field)
+        numpy.testing.assert_array_equal(attrs, expected)
 
-    standard_th = f.getth(0, 0, mkempty())
-    assert f.getfield(standard_th, 0, field) == 286331153
+        f.field_foreach(attrs, traceheader_index, indices, field)
+        numpy.testing.assert_array_equal(attrs, expected)
 
-    extension1_th = f.getth(0, 1, mkempty())
-    assert f.getfield(extension1_th, 0, field) == 572662306
+        f.putfield(traceheader, traceheader_index, field, 42)
+        f.putth(trace_index, traceheader_index, traceheader)
 
-    proprietary_th = f.getth(0, 2, mkempty())
-    assert f.getfield(proprietary_th, 0, field) == 858993459
+        traceheader = f.getth(trace_index, traceheader_index, mkempty())
+        assert f.getfield(traceheader, traceheader_index, field) == 42
 
-    f.putfield(extension1_th, 0, field, 42)
-    f.putth(0, 1, extension1_th)
+    f = get_instance_segyfile(tmpdir, 'trace-header-extensions.sgy')
+
+    standard_index = 0
+    extension1_index = 1
+    proprietary_index = 2
+    nonexistent_index = 3
+
+    expected = {
+        standard_index: [0x11111111, 0x44444444],
+        extension1_index: [0x22222222_22222222, 0x55555555_55555555],
+        proprietary_index: [0x3333, 0x6666],
+    }
```

Comment:
We could consider making the [test parametrized](https://docs.pytest.org/en/7.1.x/example/parametrize.html) and having the `header_index` and the expected output as paremeters.